sycl/source/detail/device_info.hpp

```diff
@@ -234,24 +234,23 @@ template <> struct get_device_info<bool, info::device::queue_profiling> {
 };
 
 // Specialization for atomic64 that is necessary because
-// PI_DEVICE_INFO_ATOMIC_64 isn't implemented for backend other than cuda.
-// TODO the if-statement can be removed when the other backends support
-// PI_DEVICE_INFO_ATOMIC_64.
+// PI_DEVICE_INFO_ATOMIC_64 is currently only implemented for the cuda backend.
 template <> struct get_device_info<bool, info::device::atomic64> {
   static bool get(RT::PiDevice dev, const plugin &Plugin) {
 
     bool result = false;
 
-    platform plt =
-        get_device_info<platform, info::device::platform>::get(dev, Plugin);
-
-    if (plt.get_backend() == backend::cuda) {
-      Plugin.call<PiApiKind::piDeviceGetInfo>(
-          dev, pi::cast<RT::PiDeviceInfo>(info::device::atomic64),
-          sizeof(result), &result, nullptr);
+    RT::PiResult Err = Plugin.call_nocheck<PiApiKind::piDeviceGetInfo>(
+        dev, pi::cast<RT::PiDeviceInfo>(info::device::atomic64), sizeof(result),
+        &result, nullptr);
+    if (Err == PI_INVALID_VALUE) {
+      std::cout
```

Comment:
SYCL RT shouldn't print this to the screen, just silently return false.

Reply:
OK I will update it, thanks.